### ISSUE TITLE
maptexture_t layout potentially undefined

### ISSUE THREAD

**Issue by coelckers** [fabiangreffrath/woof#1189]
`typedef struct
{
  char       name[8];
  boolean    masked;
  short      width;
  short      height;
  char       pad[4];       // unused in Doom but might be used in Boom Phase 2
  short      patchcount;
  mappatch_t patches[1];
} maptexture_t;
`

I got a problem with this when compiling as C++, not C, where 'boolean' maps to the native 1-byte 'bool' type which broke the layout of this struct. I think it's preferable to use a better defined type for the 'masked' field to avoid similar problems in the future.

**Comment by fabiangreffrath** [comment]
Sure, I can fix that. Thanks for the report! May I ask why you compile Woof with a C++ compiler?

**Comment by coelckers** [comment]
>  May I ask why you compile Woof with a C++ compiler?

Doing some experiments where I needed to put the code in a namespace which C does not have.

Aside from lack of const use in the original MBF this and a few boolean variables being set to other values than 0 or 1 were the only genuine issues I found. I'll have to compile a list of those variables, because even though it works in C I'd still consider it potentially problematic.
